### ISSUE TITLE
Overfitting on validation loss okay?

### ISSUE THREAD

**Issue by wkvong** [mlfoundations/open_clip#31]
I noticed that the CLIP validation loss curve begins to slope upwards about halfway through training on Conceptual Captions (~ epoch 15) from the figure [here](https://github.com/mlfoundations/open_clip/blob/main/docs/clip_conceptual_captions.md), but validation recall continues to increase until the end of training (epoch 30). 

Does this mean that when doing contrastive training, the procedure for early stopping be based on the validation recall performance, rather than the validation loss, since they are not necessarily tied to one another like in standard supervised learning?

**Comment by carlini** [comment]
I've seen the same thing on both CC and YFCC. (Even more surprising, to me, is that ImageNet accuracy goes up along with validation recall.) It's an interesting effect and don't have a good explanation for the cause. Maybe someone else does though.

**Comment by gabrielilharco** [comment]
In general I wouldn't recommend early stopping based on validation loss when using cross-entropy, it's not uncommon for both loss and accuracy---or other metrics you care about---to go up simultaneously after some point in training. I think the reason is that models get more confident in their predictions as training progresses, and when they are confidently wrong, this dominates the cross entropy loss.
